BinaryTree.java

```diff
@@ -15,99 +24,133 @@ public BinaryTree(BinaryTreeNode<T> root) {
     }
 
     // Check if the key exists in the tree
-    public Boolean isExist(T key) {
-        return isExist(root, key);
+    public BinaryTreeNode exists(T key) {

```

Comment:
In this form it should be something like `findNode`, since it returns something rather than boolean flag that would say if something exists.
I think even better name would be something like `containsKey`, returning boolean. `tree.exists()` feels more like checking if the tree exists rather than whether it contains given key.